apps/settings/js/panels/homescreens_list/homescreens_list.js

```diff
@@ -74,13 +84,33 @@ define(function(require) {
      * @private
      */
     _radioClick: function hl_radioClick(index) {
-      var app = this._apps[index];
-      const settings = window.navigator.mozSettings;
+      var manifestURL = this._apps[index].manifestURL;
 
       // Change the current home screen.
-      settings.createLock().set({
-        'homescreen.manifestURL': app.manifestURL
-      });
+      if (this.manifestURL === VERTICALHOME_MANIFEST) {
+        // If the current homescreen is verticalhome, we want to run the
```

Comment:
by moving these comments to above of the `if (this.manifestURL...` sentence, it might be explain the code more properly